Build failure: toppler

### Steps To Reproduce

Steps to reproduce the behavior:
1. build toppler

### Build log
https://hydra.nixos.org/build/258574948

fails when running `povray`.  `pov.log` contains the error.
```
( cd _build/cross_pov && povray ../../datafile/cross_pov/cross.ini +L../../datafile/cross_pov -D 2>> ../pov.log )
make: *** [Makefile:119: _build/cross.dat] Error 1
```
<details><summary>pov.log</summary>

```
povray: cannot open the user configuration file /build/tmp.BAbCEZwDJx/.povray/3.8/povray.conf: No such file or directory
Persistence of Vision(tm) Ray Tracer Version 3.8.0-x.10064738.unofficial (g++
 13.2.0 @ x86_64-pc-linux-gnu)
This is an unofficial version compiled by:
 'nix'
 The POV-Ray Team is not responsible for supporting this version.

POV-Ray is based on DKBTrace 2.12 by David K. Buck & Aaron A. Collins
Copyright 1991-2019 Persistence of Vision Raytracer Pty. Ltd.

Primary Persistence of Vision(tm) Ray Tracer Version 3.8.0-x.10064738.unofficial
 (g++ 13.2.0 @ x86_64-pc-linux-gnu) Architects/Developers: (Alphabetically)
  Chris Cason         Thorsten Froehlich  Christoph Lipka   

With Assistance From: (Alphabetically)
  Nicolas Calimet     Jerome Grimbert     James Holsenback    Christoph Hormann 
  Nathan Kopp         Juha Nieminen       William F. Pokorny

Past Contributors: (Alphabetically)
  Steve Anger         Eric Barish         Dieter Bayer        David K. Buck     
  Nicolas Calimet     Chris Cason         Aaron A. Collins    Chris Dailey      
  Steve Demlow        Andreas Dilger      Alexander Enzmann   Dan Farmer        
  Thorsten Froehlich  Mark Gordon         Jerome Grimbert     James Holsenback  
  Christoph Hormann   Mike Hough          Chris Huff          Kari Kivisalo     
  Nathan Kopp         Lutz Kretzschmar    Christoph Lipka     Jochen Lippert    
  Pascal Massimino    Jim McElhiney       Douglas Muir        Juha Nieminen     
  Ron Parker          William F. Pokorny  Bill Pulver         Eduard Schwan     
  Wlodzimierz Skiba   Robert Skinner      Yvo Smellenbergh    Zsolt Szalavari   
  Scott Taylor        Massimo Valentini   Timothy Wegner      Drew Wells        
  Chris Young       

Other contributors are listed in the documentation.

Support libraries used by POV-Ray:
  ZLib 1.3.1, Copyright 1995-2012 Jean-loup Gailly and Mark Adler
  LibPNG 1.6.43, Copyright 1998-2012 Glenn Randers-Pehrson
  LibJPEG 6b, Copyright 1991-2016 Thomas G. Lane, Guido Vollbeding
  LibTIFF 4.6.0, Copyright 1988-1997 Sam Leffler, 1991-1997 SGI
  Boost 1.81, http://www.boost.org/

Dynamic optimizations:
  CPU detected: AMD,FMA4
  Noise generator: generic (portable)


Rendering frame 1 of 240 (#0)

Rendering frame 1 of 240 (#0)


Parser Options
  Input file: cross.pov
  Remove bounds........On 
  Split unions.........Off
  Library paths:
    /nix/store/x3w6q10l042nh9wmzs8xy5hawdwpz0ds-povray-3.8.0-x.10064738/share/po
vray-3.8
    /nix/store/x3w6q10l042nh9wmzs8xy5hawdwpz0ds-povray-3.8.0-x.10064738/share/po
vray-3.8/ini
    /nix/store/x3w6q10l042nh9wmzs8xy5hawdwpz0ds-povray-3.8.0-x.10064738/share/po
vray-3.8/include
    ../../datafile/cross_pov
Animation Options
  Initial Frame:        0  Final Frame:      239
  Frame Step:           1
  Initial Clock:    0.000  Final Clock:    2.000
  Cyclic Animation.....On   Field render.........Off  Odd lines/frames.....Off
Image Output Options
  Image resolution.....32 by 32 (rows 1 to 32, columns 1 to 32).
  Output file..........cross000.png, 24 bpp PNG
  Dithering............Off
  Graphic display......Off
  Mosaic preview.......Off
  Continued trace......Off
Information Output Options
  All Streams to console..........On 
  Debug Stream to console.........On 
  Fatal Stream to console.........On 
  Render Stream to console........On 
  Statistics Stream to console....On 
  Warning Stream to console.......On 
==== [Parsing...] ==========================================================
File '../../datafile/cross_pov/../sprites_pov/environment.pov' line 16: Parse
 Error: Viewing angle has to be smaller than 180 degrees.
Fatal error in parser: Cannot parse input.
Render failed

```
</details>

### Additional context

Was just looking at issue for ZHF but unlikely to continue.

### Notify maintainers

@fgaz 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

NA


---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -11970,7 +11970,11 @@ with pkgs;
 
   postscript-lexmark = callPackage ../misc/drivers/postscript-lexmark { };
 
-  povray = callPackage ../tools/graphics/povray { };
+  povray = callPackage ../tools/graphics/povray {
+    # https://github.com/POV-Ray/povray/issues/460
+    # https://github.com/NixOS/nixpkgs/issues/311017
+    stdenv = gcc12Stdenv;
+  };
 
   power-profiles-daemon = callPackage ../os-specific/linux/power-profiles-daemon { };
 

